### DISCUSSION TITLE
New in pre-release: command palette improvements

### DISCUSSION THREAD

**Discussion by kjk** [sumatrapdfreader/sumatrapdf#4407] [Announcements]
I've made an improvement to `Ctrl + K` ([Command Palette](https://www.sumatrapdfreader.org/docs/Command-Palette)) in latest [pre-release](https://www.sumatrapdfreader.org/prerelease).

Previously default view was combining tabs, history files and commands.

In retrospect I think it was confusing so now those views are separate. You switch between those tabs with the first character in edit field:
* `#` for file history (all files you've opened in the past)
* `@` tabs (all tabs from all windows)
* `>` or nothing for commands

I've also added buttons at the top, below edit field, to switch between those modes. They really are to make it more obvious it's possible, you'll be better of just typing those characters.

Here's a picture of update command palette:

![image](https://github.com/user-attachments/assets/b1aa3d3b-1840-4975-98e4-03a203d84383)

**Comment by Chivos** [comment]
Is it possible to make it memorize the last selected tab (commands, history, open tabs)? If it's not stored permanently in settings.txt, at least during the session ?

**Comment by kjk** [comment]
I'm not sure that would be expected behavior.

In latest [pre-release](https://www.sumatrapdfreader.org/prerelease) you can get list of tabs with `Ctrl + Tab`.

You can bind `Ctrl + Shift + K` (`Ctrl + H`) for history:
```
Shortcuts [
    [
        Cmd = CmdCommandPalette #
        Key = Ctrl + Shift + K
    ]
```

I might add this as built-in shortcut

https://www.sumatrapdfreader.org/docs/Customizing-keyboard-shortcuts
https://www.sumatrapdfreader.org/docs/Commands#cmdcommandpalette

**Reply by Chivos** [reply]
Current default behavior is to open command palette in the Commands part.
If I switch to File History to acces a file, close and reopen the command palette, it goes back to the Commands part of the palette. I thought that reopening the palette in the File History section would have been the expected behavior in that case.

However, binding differents shortcut is a good alternative !

**Reply by kjk** [reply]
You can even change the default behavior to file history:
```
Shortcuts [
    [
        Cmd = CmdCommandPalette #
        Key = Ctrl + K
    ]
]
```
